agent/__init__.py

```diff
@@ -390,6 +392,8 @@ def send_ping(dev=False):
         'confinement': CONFINEMENT.name,
         'installation': detect_installation().name
     }
+    if ping.get('deb_packages_hash') != packages['hash']:
+        payload['deb_packages'] = packages
```

Comment:
Pls consider my recommendation to always send something in `payload['deb_packages']`